pkgs/development/libraries/glfw/3.x.nix

```diff
@@ -6,17 +6,17 @@
 }:
 
 stdenv.mkDerivation rec {
-  version = "3.3.4";
+  version = "3.3.5";
   pname = "glfw";
 
   src = fetchFromGitHub {
     owner = "glfw";
     repo = "GLFW";
     rev = version;
-    sha256 = "sha256-BP4wxjgm0x0E68tNz5eudkVUyBnXkQlP7LY3ppZunhw=";
+    sha256 = "sha256-1KkzYclOLGqiV1/8BsJ3e+pXMQ6a+sjLwZ7mjSuxxbA=";
   };
 
-  patches = lib.optional waylandSupport ./wayland.patch;
+  patches = if waylandSupport then ./wayland.patch else ./x11.patch;
```

Comment:
please add a comment about the necessity of the patch.
Can a user need both patches at the same time ? (for example when targetting XWayland ? I'm not familiar with glfw, it might not make sense.)

Note that would only minimally increase closure size as wayland support adds a dependency on libxkbcommon which brings libx11 and libxcb in already.

Reply:
> please add a comment about the necessity of the patch.

Done!

> Can a user need both patches at the same time? (for example when targetting XWayland?)

IIUC the Wayland patch is apply for a file that's only compiled when Wayland support is specified and the X11 patch only affects linking instructions for building for X11, and XWayland is for running the X11 build on Wayland, so the two patches should be mutually exclusive.